GitUI/UserControls/RevisionGrid/RevisionGridRefRenderer.cs

```diff
@@ -32,7 +33,7 @@ public static void DrawRef(bool isRowSelected, Font font, ref int offset, string
             var rect = new Rectangle(
                 bounds.X + offset,
                 bounds.Y + outerMarginTopBottom,
-                textSize.Width + arrowWidth + paddingLeftRight + paddingLeftRight - 1,
+                Math.Min(bounds.Width - offset, textSize.Width + arrowWidth + paddingLeftRight + paddingLeftRight - 1),
```

Comment:
TODO? Added to list/issues for v3-ux?
It should be more clear that branches/tags are clipped.
Also, the sort order in #5077 by @jbialobr is no longer followed (tags are displayed first, tracked remote are displayed after all local branches)

Order in tooltip is not as in the grid
The tooltip has no color, and should explain local/remote branches, tags etc.

![image](https://user-images.githubusercontent.com/6248932/42788707-589fa5c4-8961-11e8-86f7-81999080d8d2.png)

Reply:
@gerhardol thank you very much for the review.

- I've reinstated the ordering from #5077 
- The tool tip now has the same ordering as the ref badges in the cell
- Ref badges no longer render under the multiline message indicator

![image](https://user-images.githubusercontent.com/350947/42811355-468af0cc-89b2-11e8-83b9-cd9bbf5e407a.png)

Other items:

- I'm not sure how to better show when ref badges are clipped. Do you or anyone else have a suggestion?
- Colour tool tips would be great, but a bit of a nightmare in WinForms. Another point for WPF.

(Will push these fixes today)

Reply:
> I'm not sure how to better show when ref badges are clipped. Do you or anyone else have a suggestion?

A separate overflow badge would be slightly more clear. But this is good enough right now

Thanks for all work with this PR. I am still voting to merge this now and take comments over time (and let you start a new test branch...).